### DISCUSSION TITLE
Fail to record voice via ASIO4ALL

### DISCUSSION THREAD

**Discussion by schjr** [sist-cs120/project-wiki#10] [Project 1 Q&A]
I try to run the AudioLatencyDemo under the guidence of the instructions. However, the Demo cannot record voice using the microphone and ASIO4ALL. Surprisingly, when I press the "Test Latency" button, some buzz comes out from the speaker and there is some wave on the recording screen.
Then I ran the AudioRecordingDemo and it failed to record anything.
My microphone works since when I accidently used Windows's audio interface to run the AudioLatencyDemo, the demo could record sound in the environment although it failed to test latency. Such thing happens only when I use the ASIO.
The player demo works and it can play sound given a frequency.
![2021-09-29 (2)](https://user-images.githubusercontent.com/53388741/135285478-87cdeab6-0dfb-4027-bca4-85660e3ebe68.png)

2021/10/2:
Thanks for all the suggestions. The normal setting of ASIO4ALL only uses the sound card's output and stereo mixer. The mic is closed by defualt. However, it turns out that when I open the mic and close the stereo mixer (If I do not close the mixer, the mic cannot be used), the mic receives unbearable high background noise and It could not receive the buzz sound generated by the demo. Other kinds of ASIO drivers I found all have various problems. I am still trying to find a solution :) 
(No SST device is found in the control panel of ASIO4ALL)

**Comment by JohnnySIST** [comment]
I've met the same problem too.
![image](https://user-images.githubusercontent.com/72036531/135297792-56269844-22e5-4b25-bf03-23aef5089557.png)
My recording device is **Realtek High Definition Audio(SST)**. I think the reason might be the poor compatibility between ASIO4ALL and Intel(R) SST, since in ASIO4ALL VERSION 2.15 (BETA 1) released on 11 MARCH 2021 the support for Intel(R) SST was just added. So I turned to another ASIO driver [https://github.com/SunbossRS/FLStudioASIO](url). It worked well, but I haven't tested the latency.

**Reply by ghost** [reply]
The latency can be tested with a loop test. Write one or use some reliable tools (e.g, https://oblique-audio.com/rtl-utility.php)

**Reply by ghost** [reply]
I think most audio cards should have the ASIO compatibility, but there might be issues in driver implementation.  If the issue remains, reach TAs to borrow USB sound cards that have been tested and verified by me.

**Comment by ghost** [comment]
> I try to run the AudioLatencyDemo under the guidence of the instructions. However, the Demo cannot record voice using the microphone and ASIO4ALL. Surprisingly, when I press the "Test Latency" button, some buzz comes out from the speaker and there is some wave on the recording screen. Then I ran the AudioRecordingDemo and it failed to record anything. My microphone works since when I accidently used Windows's audio interface to run the AudioLatencyDemo, the demo could record sound in the environment although it failed to test latency. Such thing happens only when I use the ASIO. The player demo works and it can play sound given a frequency. ![2021-09-29 (2)](https://user-images.githubusercontent.com/53388741/135285478-87cdeab6-0dfb-4027-bca4-85660e3ebe68.png)

Have you correctly selected the recording device in the ASIO driver? Some devices have multiple audio input interfaces. You must specify one correctly. And also try to use the latency test tool I mentioned to ensure the ASIO driver works.

**Comment by yhyan96** [comment]
According to my experience, if the devices cannot play or record any sound, as Johnny showed, please first check the ASIO control panel. Refer to the [instruction manual](https://www.asio4all.org/index.php/help/instruction-manual/). If this issue is caused by SST, the [SST ISSUE](https://www.asio4all.org/index.php/help/sst-issue/) may be helpful. You can also try another ASIO driver as Johnny recommends or another sound device without SST.
